README.md

<!-- README.md is generated from README.Rmd. Please edit that file -->

# covid19ita

<img src='man/figures/logo-laims.jpg' align='right' height='120' /><img src='man/figures/logo-ubep.png' align='right' height='120' />

<!-- badges: start -->

[![R build
status](https://github.com/UBESP-DCTV/covid19ita/workflows/R-CMD-check/badge.svg)](https://github.com/UBESP-DCTV/covid19ita/actions)
[![Codecov test
coverage](https://codecov.io/gh/UBESP-DCTV/covid19ita/branch/master/graph/badge.svg)](https://codecov.io/gh/UBESP-DCTV/covid19ita?branch=master)

[![CRAN
status](https://www.r-pkg.org/badges/version/covid19ita)](https://CRAN.R-project.org/package=covid19ita)
[![Lifecycle:
maturing](https://img.shields.io/badge/lifecycle-maturing-blue.svg)](https://www.tidyverse.org/lifecycle/#maturing)

[![CII Best
Practices](https://bestpractices.coreinfrastructure.org/projects/3892/badge)](https://bestpractices.coreinfrastructure.org/projects/3892)
<!-- badges: end -->

Lo scopo del pacchetto `{covid19ita}` è quello di fornire una
piattaforma per il monitoraggio della diffusione dell’infezione da
CoViD-19 in Italia.

The goal of `{covid19ita}` is to provide a platform for the monitoring
of COVID-19 infection diffusion in Italy.

## Installazione/Installation

È possibile installare la versione di sviluppo di `{covid19ita}` dal
sorgente su [GitHub](https://github.com/) tramite l’istruzione:

You can install the development version of `{covid19ita}` from the
source on [GitHub](https://github.com/) with:

``` r
# install.packages("remotes")
remotes::install_github("UBESP-DCTV/covid19ita")
```

> Nota per poter utilizzare gli accessi riservati sulla propria
> installazione locale e richiesto il pacchetto `{covid19.icuve}`
> riservato all’uso esclusivo di chi ne ha diritto. Nel caso ci si
> trovasse in tale situazione e si volesse una installazione completa
> utilizzare l’istruzione seguente.
>
> ``` r
> remotes::install_github("UBESP-DCTV/covid19.icuve")
> remotes::install_github("UBESP-DCTV/covid19ita")
> ```
>
> Se autorizzati, il sistema completo verrà installato/aggiornato
> regolarmente.

## Shiny App

<img src='inst/app/www/covid19ita_screen.png' align="center"/>

**Ultimo aggiornamento/Last app update**: 2020-12-30 22:55:10.

> È possibile visitare la Shiny App `covid19ita` ospitata e operante sui
> server UBEP ([Unità di Biostatistica, Epidemiologia e Sanità
> Pubblica](https://ubesp.jimdofree.com/)) [**QUI (versione
> italiana)**](https://r-ubesp.dctv.unipd.it/shiny/covid19ita/)

È inoltre possibile eseguire una versione dell’app locale sul proprio
computer tramite:

> You can visit the `covid19ita` Shiny App hosted and running on the
> UBEP ([Unit of Biostatistics, Epidemiology, and Public
> Health](https://ubesp.jimdofree.com/)) servers [**HERE (English
> version)**](https://r-ubesp.dctv.unipd.it/shiny/covid19italy/).

You can also run a local version of the app from your computer with:

``` r
library(covid19ita)
run_app()
```

## Data

Il pacchetto `{covid19ita}` contiene le versioni R delle tre [basi di
dati ufficiali](https://github.com/pcm-dpc/COVID-19/) con dettaglio a
livello nazionale, regionale e provinciale.

The `{covid19ita}` package provide R versions of the three [official
Italian COVID-19 datasets](https://github.com/pcm-dpc/COVID-19/) at
national, regional and provincial level of detail.

**Ultimo aggiornamento dati/Last data update**: 2021-01-03 00:08:45.

È possibile accedere ai dati tramite:

Access the data with:

``` r
library(covid19ita)
data(dpc_covid19_ita_andamento_nazionale) # national level data
data(dpc_covid19_ita_regioni)             # regional level data
data(dpc_covid19_ita_andamento_province)  # provincial level data
```

È possibile accedere anche alla documentazione dei dati in uno dei
seguenti modi:

You can access to the data documentation with any of:

``` r
?dpc_covid19_ita_andamento_nazionale
?dpc_covid19_ita_regioni
?dpc_covid19_ita_province

?dpc_covid19_ita
```

## Attribuzione dei dati/Data attribution

**Licenza/License**:
[CC-BY-4.0](https://creativecommons.org/licenses/by/4.0/deed.en) -
[Visualizza licenza/View
license](https://github.com/pcm-dpc/COVID-19/blob/master/LICENSE)

**Autore dei dati/Dataset author/editor**: [Dipartimento della
Protezione Civile](http://www.protezionecivile.it/)

**Fornitore dei dati/Data provided by**: [Ministero della
Salute](http://www.salute.gov.it/)

**Dati originariamente gestiti e processati da/Original data processing
and management by**: [Dipartimento della Protezione
Civile](http://www.protezionecivile.it/)

## Richiesta di funzionalità/Feature request

Per richiedere funzionalità aggiuntive è possibile aprire un
[issue](https://github.com/UBESP-DCTV/covid19ita/issues).

If you need some more features, please open an
[issue](https://github.com/UBESP-DCTV/covid19ita/issues).

## Bug reports

Nel caso in cui si incontrasse un bug nelle funzioni del pacchetto, per
favore si riporti un [reprex](https://github.com/tidyverse/reprex)
all’interno di un
[issue](https://github.com/UBESP-DCTV/covid19ita/issues).

If you encounter a bug in the package’s functions, please file a
[reprex](https://github.com/tidyverse/reprex) (minimal reproducible
example) in a [issue](https://github.com/UBESP-DCTV/covid19ita/issues).

Se si incontra un bug nell’applicazione, aprire un
[issue](https://github.com/UBESP-DCTV/covid19ita/issues) descrivendo: -
Passaggi per riprodurre/vedere il problema - Descrizione del problema
dettagliata - Descrizione di cosa ci si sarebbe aspettato

If you encounter a bug in the app, please describe: - The steps to
reproduce/visualize the problem - The problem in detail - The behavior
you expected

## Codice di condotta/Code of Conduct

Il progetto `covid19ita` è sviluppato con un [Codice di condotta per chi
contribuisce](CODE_OF_CONDUCT.md). Dando il tuo contributo a questo
progetto, accetti di rispettarne i termini.

Please note that the `covid19ita` project is released with a
[Contributor Code of Conduct](CODE_OF_CONDUCT.md). By contributing to
this project, you agree to abide by its terms.


NEWS.md
# covid19ita 0.119.1

# covid19ita 0.119.0

# covid19ita 0.118.0

# covid19ita 0.117.0

* improved loading time making activation on tab visualization only
* improved map server module
* Add Azero module

# covid19ita 0.116.0

* fix occurences of `group_by_*()` using `group_by(across())`
* update utils_tsicuve to correctly manage periodic time series
* Update timeseries to manage 2021 year, introducing a periodicity of
  a week.
* Fix forecasting lowerbounding it to zero, and limiting it to 7 days
  ahead (#92).
* Ability to selecting the measures for national ts (#91).
* Add `Theme analyses` tab (#91).
* old `focus` items in `archive` tab (#91).
* reshape sidebar (#91).

# covid19ita 0.115.0

* Updated Log in system to automatically login to the public user and
  allowing for sign in with other users after the visualization of
  the dashboard.

# covid19ita 0.114.1

* update `Dockerfile`
* added delay for the shock in the exogen model
* data update

# covid19ita 0.114.0

* added ICU exogen model with shock on non critical hospitalizations

# covid19ita 0.113.0

* Added `mod_reg_tsicuve_noncritical` to show timeseries and the 
  corresponding ETS forecasting for non-critical hospitalizations.

# covid19ita 0.112.0

* Added ets prediction to siterep (including estimation up to 15 days
  and CI95% up to 4 days).
* added provincial and center level at siterep.

# covid19ita 0.111.0

* Added `mod_reg_tsicuve` to section *Terapie intensive*
* Moved module `mod_ts_icuve`to section "In evidenza"
* Fix non-ASCII characters
* Update WORDLIST
* Move `{covid19.icuve}` into **Enhances** to avoid its installation 
  even when installed with `dependencies = TRUE` (solves #89).

# covid19ita 0.110.0

* Remove `ylim` in `ggbeds` in `mod_icuve_ts`
* Solved the problem with `mod_icuve_ts` at line 186 (#86)
* Solved the problem with default account (#86)
* Solve automatic update of Figure 3 in `mod_icuve_ts` (#86)
* Fixed 1/2 of #85
* Fixed problem with `ggswab` (#84)
* data update
* Integrated login credential in the main app
* New module `mod_tsicuve` that shows forecasts of Time-Series models
  applied to ICU data from the Veneto region. The following Time-Series
  models were considered: Holter-Winters Filtering (HTF), Exponential 
  smoothing state space model (ETS), and ARIMA.
* Added `xts`, `tscount`, and `forecast` to Imports

# covid23icuve 0.109.1

* Restore home page after login
* data update
* Added login module.
* Added modules `mod_icuve_sitrep`, `mod_icuve_ts` and `mod_icuve_static` to showing ICU's
  situation report, indicators on time series and static data, and predictions for ICUs in Veneto.

# covid19ita 0.109.0

* data update

# covid19ita 0.108.0

* data update

# covid19ita 0.107.1

* Update default visualization for regions: total current cases, and
  total current patients in ICUs.
* reformat x scales for ts-plots using 2 weeks instead of 1 day.
* update `data-UPDATE.R` script to automate committing of updates
* data update

# covid19ita 0.95.0 -- 0.106.0

* data update

# covid19ita 0.94.0

* required dev version of tibble due to an error
* update data
* package `{growthcurve}` is nomore into CRAN, switched to the
  development version from GitHub (briandconnelly/growthcurve)

# covid19ita 0.93.0

* fix update function
* update data

# covid19ita 0.92.0

* update updating functions
* adjust naming convention to the new dpc ones
  (https://github.com/pcm-dpc/COVID-19/blob/master/CHANGELOG_EN.md)
* data update

# covid19ita 0.90.0 -- 0.91.0

* data update

# covid19ita 0.89.0

* data update
* reformat default set o measure displayed

# covid19ita 0.62.0 -- 0.88.0

* data update

# covid19ita 0.61.0

* update gh-action with updated dependencies
* data update

# covid19ita 0.60.0

* data update

# covid19ita 0.59.0

* update remote gh-action (`{lintr}` was out-dated)
* data update

# covid19ita 0.56.0 -- 0.58.0

* data update

# covid19ita 0.55.0

* changed `readr::write_rds()` with `saveRDS()` due to the error
  "simpleWarning: cannot initialize lzma encoder, error 5"
* data update

# covid19ita 0.53.0 -- 0.54.0

* data update

# covid19ita 0.52.0

* data update
* slider 7 day window incomplete tails' data fixed for maps module

# covid19ita 0.47.0 -- 0.51.0

* data update

# covid19ita 0.46.0

* data update (including case tested)

# covid19ita 0.45.0

* improve maps section
* data update

# covid19ita 0.44.0

* data update
* Add name (and email) for GitHub Action User, to be used in its
  git commits (#64).
* separate CI checks into separate workflow for master and develop
  branches. Push/pull-request to master branch will perform full checks
  including oldrel and dev R version. Push/pull-request on develop
  branch will check only against release R version (#65)

# covid19ita 0.43.0


* Update data
* Included covr into gh-action to avoid badges conflicts
* Activate gh-action for (R3.5, R3.6)*(Linux, OSX, Windows) + Rdevel on
  OSX (the only one available at the moment for gh-actions) (#59)
* Remove Travis and Appveyor CI (#59)

# covid19ita 0.42.0

* added cii best practice badge
* update data
* activate lintr on Travis-CI and GitHub Actions
* fix lintr check (all but `extraction_operator_linter` (1408 problems),
  `line_length_linter` (913 problems), and `cyclocomp_linter`
  (2 problems in *_mod_maps.R)).
* activate GitHub Actions for CI.
* activate pkgdown, including a GitHub action to automatically deploy
  the documentation's WEB site.
* update README including information on how to contribute
  (resolve #56).
* translated 20200415_tamponi (close #54).
* refactored 20200415_tamponi (#54).

# covid19ita 0.41.0

* included focus on "Tests and hospitalization"
* data update
* fix text in 20200404_magnani and 20200406_mort_veneto for ita and eng

# covid19ita 0.40.0

* data update

# covid19ita 0.39.0

* data update
* fix map section in eng (#49)

# covid19ita 0.38.0

* update data dpc
* `README` update (#4)

# covid19ita 0.37.0


R/eng-app_ui.R
#' The application User-Interface
#'
#' @param request Internal parameter for `{shiny}`.
#'     DO NOT REMOVE.
#' @import shiny
#' @import shinydashboard
#' @noRd
eng_app_ui <- function(request) {
  tagList(

    # Leave this function for adding external resources
    eng_golem_add_resource(),

    dashboardPage(
      title = " COVID-19 - Italy ",
      # Helpers for dashboard's header and sidebar
      eng_dashboard_header(), eng_dashboard_sidebar(), eng_dashboard_body(),
      # Main dashboard skin colour
      skin = "red"
    )
  )
}



#' Header's dashboard
#'
#' Helper function defining the header of the dashboard
#'
#' @import shiny
#' @importFrom shinydashboard dashboardHeader
#' @noRd
eng_dashboard_header <- function() {
  dashboardHeader(
    title = "COVID-19 - Italy",

    dropdownMenu(
      type = "messages",
      messageItem(
        from = "Highlights",
        message = "Impact of testing on hospitalizations",
        icon = icon("search")
      ),
      messageItem(
        from = "Data",
        message = glue::glue("Data updated ({last_data_update})"),
        icon = icon("database")
      ),
      messageItem(
        from = "Lingua",
        message = "Clicca qui per il sito in italiano",
        icon = icon("flag"),
        href = "https://r-ubesp.dctv.unipd.it/shiny/covid19ita/"
      )
    )
  )
}

#' Sidebars dashboard
#'
#' Helper function defining the sidebar of the dashboard
#'
#' @import shiny
#' @importFrom shinydashboard dashboardSidebar sidebarMenu menuItem
#' @noRd
eng_dashboard_sidebar <- function() {
  dashboardSidebar(sidebarMenu(
    id = "sidebar",

    eng_mod_help_plot_ui("help"),

    menuItem("Home", tabName = "home", icon = icon("home")),

    menuItem("Highlights",
      icon = icon("bullseye"),

      menuSubItem(
        text = "2020-04-15 Effects of testing", tabName = "20200415TampHosp",
        icon = icon("flag")
      ),

      menuSubItem("2020-04-06 Veneto Mortality",
        tabName = "20200406MortVeneto",
        icon = icon("flag")
      ),
      menuSubItem("2020-04-05 Mortality",
        tabName = "20200405Mort",
        icon = icon("flag")
      ),
      menuSubItem("2020-03-31 Comparative",
        tabName = "20200331Comp",
        icon = icon("flag")
      ),
      menuSubItem("2020-03-28 Veneto",
        tabName = "20200328Tot_hosp",
        icon = icon("flag")
      ),
      menuSubItem("2020-03-25 Veneto",
        tabName = "20200325Hosp",
        icon = icon("flag")
      ),
      menuSubItem("2020-03-24 Methodology",
        tabName = "20200323Picco",
        icon = icon("flag")
      ),
      menuSubItem("2020-03-21 Alessandria",
        tabName = "20200321Alessandria",
        icon = icon("flag")
      ),
      menuSubItem("2020-03-21 Vercelli",
        tabName = "20200321Vercelli",
        icon = icon("flag")
      ),
      menuSubItem("2020-03-20 Novara",
        tabName = "20200320Novara",
        icon = icon("flag")
      ),
      menuSubItem("2020-03-19 Veneto",
        tabName = "20200319Veneto",
        icon = icon("flag")
      ),
      menuSubItem("2020-03-18 Piemonte",
        tabName = "20200318Piemonte",
        icon = icon("flag")
      ),
      menuSubItem("2020-03-18 FVG",
        tabName = "20200318Fvg",
        icon = icon("flag")
      ),
      menuSubItem("2020-03-14 Veneto",
        tabName = "20200314Veneto",
        icon = icon("flag")
      )
    ),

    menuItem("Epidemic",
      icon = icon("chart-line"),
      menuSubItem("National", tabName = "national", icon = icon("flag")),
      menuSubItem("Regional", tabName = "regional", icon = icon("map")),
      menuSubItem("Provincial", tabName = "provincial", icon = icon("location-arrow"))
    ),

    menuItem("Principal indices", tabName = "impact", icon = icon("compass")),

    menuItem("Maps", tabName = "geo_spatialTot", icon = icon("map-marked-alt")),

    menuItem("Issues",
      icon = icon("exclamation-triangle"),
      href = "https://github.com/UBESP-DCTV/covid19ita/issues/"
    ),

    menuItem("Info and sources", tabName = "data_tab", icon = icon("database")),

    eng_mod_info_sidebar_ui("summary_today")
  ))
}





#' Body dashboard
#'
#' Helper function defining the main body of the dashboard
#'
#' @import shiny
#' @importFrom shinydashboard dashboardBody box
#' @noRd
eng_dashboard_body <- function() {
  dashboardBody(
    tags$head(includeScript(app_sys("app/www/google-analytics.js"))),
    tabItems(
      eng_dashboard_home_body(),

      tabItem(
        tabName = "20200415TampHosp",
        h2("Impact of testing on hospitalizations"),
        eng_mod_0415_tamponi_ui("tamp_hosp")
      ),

      tabItem(
        tabName = "20200406MortVeneto",

        h2("Preliminary analysis of the overall mortality in 122 municipalities of the Veneto Region from the 1st of March to the 21 of March 2020."),
        eng_mod_0406_mort_ve_ui("mort_veneto")
      ),
      tabItem(
        tabName = "20200405Mort",
        h2("Preliminary analysis of the overall mortality in 1084 italian municipalities from the 1st of March to the 21 of March 2020."),
        eng_mod_0404_magnani_ui("mortality")
      ),
      tabItem(
        tabName = "20200331Comp",
        h2("Comparative analysis between the Piemonte Region and Veneto Region of the epidemiological data relative to Covid-19 infection."),
        eng_mod_0331_ui("ven_pie")
      ),
      tabItem(
        tabName = "20200328Tot_hosp",
        h2("Possible effect on hospitalizations of the health policies implemented by the Veneto region"),
        eng_mod_0328_hosp_ui("tot")
      ),
      tabItem(
        tabName = "20200325Hosp",
        h2("Possible effect on hospitalizations of the health policies implemented by the Veneto region"),
        eng_mod_0325_hosp_ui("hosp")
      ),
      tabItem(
        tabName = "20200323Picco",
        h2("Impact of statistical uncertainty on COVID-19 predictions"),
        eng_mod_0323_picco_ui("picco")
      ),
      tabItem(
        tabName = "20200321Alessandria",
        h2("Expected number of total cases in Alessandria"),
        eng_mod_0320_novara_ui("da_alessandria")
      ),
      tabItem(
        tabName = "20200321Vercelli",
        h2("Expected number of total cases in Vercelli"),
        eng_mod_0320_novara_ui("da_vercelli")
      ),

      tabItem(
        tabName = "20200320Novara",
        h2("Expected number of total cases in Novara"),
        eng_mod_0320_novara_ui("da_novara")
      ),

      tabItem(
        tabName = "20200319Veneto",
        h2("Possible effect on ICU admissions of the health policies implemented by the Veneto region"),
        eng_mod_0318_intensive_ui("21")
      ),

      tabItem(
        tabName = "20200318Piemonte",
        h2("Possible effect of the health policies implemented by the Piemonte region"),
        eng_mod_0318_piemonte_ui("20200318_piemonte")
      ),
      tabItem(
        tabName = "20200318Fvg",
        h2("Possible effect of the health policies implemented by the Friuli Venezia Giulia region"),
        eng_mod_0318_friuli_ui("20200318_fvg")
      ),
      tabItem(
        tabName = "20200314Veneto",
        h1("Possible effect of the health policies implemented by the Veneto region"),
        eng_mod_0314_ui("dapb")
      ),

      tabItem(
        tabName = "national",
        h2("National events"),
        box(
          width = 12, title = "Instructions",
          p("Visualize/hide one or more measures on the graph by clicking on the items in the legend. Double click to only visualize the selected item."),
          p("Click on the autoscale button (the third one) to maximize the size of the graph."),
        ),
        eng_mod_ts_ita_ui("ts_nat_cum", title = "Time series trend - cumulative events"),
        eng_mod_ts_ita_ui("ts_nat_inc", title = "Time series trend - daily new events")
      ),


      tabItem(
        tabName = "regional",
        h2("Regional events"),
        box(
          width = 12, title = "Istructions",
          p("Add/remove computations for one or more region/measure by adding/removing it from the box."),
          p("NOTE: The number of tests per day can be visualized on the graph (by selecting the correct item in the legend) but it is hidden by default as it is off the scale (compared to the other measures)."),
          p(""),
          p("Visualize/hide one or more regions/measures on the graph by clicking on the items in the legend. Double click to only visualize the selected item."),
          p("Click on the autoscale button (the third one) to maximize the size of the graph."),
        ),

        h3("Regional time series by measure"),
        fluidRow(
          box(
            title = "Cumulative events", width = 12,
            eng_mod_ts_reg_ui("ts_reg_cum_reg")
          )
        ),

        fluidRow(
          box(
            title = "Daily new events", width = 12,
            eng_mod_ts_reg_ui("ts_reg_inc_reg")
          )
        ),


        h3("Time series by region"),
        fluidRow(
          box(
            title = "Cumulative events", width = 12,
            eng_mod_ts_reg_ui("ts_reg_cum_mes")
          )
        ),
        fluidRow(
          box(
            title = "Daily new events", width = 12,
            eng_mod_ts_reg_ui("ts_reg_inc_mes")
          )
        )
      ),


      tabItem(
        tabName = "provincial",
        h2("Provincial events"),
        p("Add/remove computations for one or more region/measure by adding/removing it from the box."),
        p("Visualize/hide one or more regions on the graph by clicking on the items in the legend. Double click to only visualize the selected item."),
        p("Click on the autoscale button (the third one) to maximize the size of the graph."),

        h3("Time series"),
        fluidRow(
          box(
            title = "Cumulative events", width = 12,
            eng_mod_ts_prv_ui("ts_prv_cum")
          )
        ),

        fluidRow(
          box(
            title = "New events", width = 12,
            eng_mod_ts_prv_ui("ts_prv_inc")
          )
        )
      ),


      tabItem(
        tabName = "data_tab",
        h2("Data Info"),
        p("This app takes the needed data from the offical records that track the italian COVID-19 outbreak, on a national, regional and provincial level"),

        p(
          "Data are processed and made available by the ",
          a(href = "http://www.protezionecivile.it/web/guest/home", target = "_blank", "Presidenza del Consiglio dei Ministri - Dipartimento di Protezione Civile"),
          " (Italian Civil Protection Department) and licensed under ", a(href = "https://creativecommons.org/licenses/by/4.0/deed.en", target = "_blank", "CC-BY-4.0"),
          "as provided by the ", a(href = "", target = "_blank", "Ministero della Salute"), "(Ministry of Health)."
        ),

        p("Data is usually updated daily at 6pm."),

        p(
          "Further information about data attribution and availability can be found at the",
          a(href = "https://github.com/UBESP-DCTV/covid19ita/", target = "_blank", "web page of covid19ita."),
          "."
        ),

        h2("Software"),
        p(
          HTML("The app <strong>covid19ita</strong> was developed using R ver. 3.6.3 as an expansion package. The source code of the package and the app is freely available online on Github at "),
          a(href = "https://github.com/UBESP-DCTV/covid19ita/", target = "_blank", "https://github.com/UBESP-DCTV/covid19ita/"), "."
        ),

        p("For the app development the following expansion packages were used: {shiny} ver. 1.4.0, {shinydashboard} v.0.7.1 and {golem} ver. 0.2.1."),

        p("The analyses were performed using functions of the following packages: {stats} ver. 3.6.3, and {gam} ver. 1.16.1"),

        p("Our graphs were produced with the following packages: {ggplot2} ver. 3.3.0 and {plotly} ver. 4.9.2."),

        h3("Note for R users"),
        p(
          "In addition to this app (that can be locally excecuted, after installing the package {covid19ita}, by running the code `run_app()`), the R package {covid19ita}, available on ",
          a(href = "https://github.com/UBESP-DCTV/covid19ita/", target = "_blank", "GitHub"),
          " and licenced under ",
          a(href = "https://creativecommons.org/licenses/by/4.0/deed.en", target = "_blank", "CC-BY-4.0"),
          ", makes raw data available for direct use."
        )
      ),


      tabItem(
        tabName = "impact",
        h1("Principal indices"),
        eng_mod_ind_ita_ui("20200315")
      ),

      tabItem(
        tabName = "geo_spatialTot",
        # h1("Mappe 1"),
        eng_mod_maps_ui("geo_1")
      )
    )
  )
}





#' Add external Resources to the Application
#'
#' This function is internally used to add external
#' resources inside the Shiny application.
#'
#' @import shiny
#' @importFrom golem add_resource_path activate_js favicon bundle_resources
#' @noRd
eng_golem_add_resource <- function() {
  add_resource_path(
    "www", app_sys("app/www")
  )

  tags$head(
    favicon(),
    bundle_resources(
      path = app_sys("app/www"),
      app_title = " {covid19ita} ",
      meta = list(
        title = "{covid19ita} App",
        description = "Monitoring platform analysis of the COVID-19 infection in Italy.",
        url = "https://r-ubesp.dctv.unipd.it/shiny/covid19ita/",
        image = app_sys("app/www/ubep_covid.png"),
        image_alt = "Emergenza COVID-19 ITA",
        twitter_creator = "@CorradoLanera",
        twitter_card_type = "summary",
        twitter_site = "@ubesppadova",
        og_locale = "it_IT",
        og_author = c(
          "Corrado L.", "Dario G.", "Paola B.", "Danila A.", "Giulia L.",
          "Ilaria P.", "Nicolas D."
        ),
        og_site_name = "covid19ita"
      )
    )
    # Add here other external resources
    # for example, you can add shinyalert::useShinyalert()
  )
}
